Misc/NEWS.d/next/Core and Builtins/2021-07-03-00-20-39.bpo-43908.YHuV_s.rst

```diff
@@ -0,0 +1,3 @@
+Types with the :const:`Py_TPFLAGS_IMMUTABLETYPE` flag set can now implement
+the :pep:`590` vectorcall protocol.  Previously, this was only available for
+:ref:`static types <static-types>`.  Patch by Erlend E. Aasland.
```

Comment:
The PEP 590 doesn't document `tp_vectorcall`, only `tp_vectorcall_offset`: https://www.python.org/dev/peps/pep-0590/#changes-to-the-pytypeobject-struct

`tp_vectorcall_offset` is usable in heap type with the special `__vectorcalloffset__` member: https://docs.python.org/dev/c-api/structures.html#c.PyMemberDef

For example, _functools.partial type became a heap type in Python 3.10 and it defines `tp_vectorcall_offset` using a `__vectorcalloffset__` member.

So I don't understand well this NEWS entry.

Reply:
Quoting from [https://www.python.org/dev/peps/pep-0590/#subclassing](PEP 590, Subclassing):
_"Heap types never inherit the vectorcall protocol because that would not be safe (heap types can be changed dynamically)."_

Reply:
Maybe _"Types with Py_TPFLAGS_IMMUTABLETYPE can now **inherit** the vectorcall protocol"_ would be better.

Reply:
The NEWS entry says that previously it was not possible to implement the vectorcall in heap type. But I understand that it was possible. It's just that previously, it was not inherited. For me, the NEWS entry is misleading.

Reply:
You're right, it is. I'll correct it.

Reply:
Fixed with 1b035eb0a748cf4560b9cd10c6ae4aa18a56621a